src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -267,6 +269,7 @@ protected virtual void Initialize()
                     throw new FormatException(string.Format("Unable to parse {0} file.", ScriptConstants.HostMetadataFileName), ex);
                 }
 
+                ScriptConfig.HostConfig.HostConfigMetadata = hostConfig;
```

Comment:
We wanted to surface the config object to the extensions, so it needs to come through here.  This is a general solution. Think of the other properties on JobHostConfig as type-specific projections from HostConfigMetadata.